test/c/eventlog_test.cpp

```diff
@@ -6,16 +6,30 @@
 
 TEST(LCM_C, EventLogBasic) {
     // Open and close a log file.
+#ifndef WIN32
```

Comment:
To minimize the use if #ifdef, suggest adding the following functions (or similar) to test/c/common.c:

    char* make_tmpnam() {
    #ifndef WIN32
      return tmpnam(NULL);
    #else
      return _tempnam(NULL, NULL);
    #endif
    }

    void free_tmpnam(char* fname) {
    #ifdef WIN32
      free(fname);
    #endif
    }

Then modify eventlog_test.cpp to always call make_tmpnam() and free_tmpnam() regardless of platform.